CondTools/SiPhase2Tracker/plugins/DTCCablingMapProducer.cc

```diff
@@ -0,0 +1,299 @@
+#include "CondFormats/SiPhase2TrackerObjects/interface/TrackerDetToDTCELinkCablingMap.h"
+#include "CondFormats/SiPhase2TrackerObjects/interface/DTCELinkId.h"
+
+
+
+
+// trim from start (in place)
+static inline void ltrim(std::string &s)
+{
+	s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int ch)
+	{
+		return !std::isspace(ch);
+	}));
+}
+
+// trim from end (in place)
+static inline void rtrim(std::string &s)
+{
+	s.erase(std::find_if(s.rbegin(), s.rend(), [](int ch)
+	{
+		return !std::isspace(ch);
+	}).base(), s.end());
+}
+
+// trim from both ends (in place)
+static inline void trim(std::string &s)
+{
+	ltrim(s);
+	rtrim(s);
+}
+
+
+
+class DTCCablingMapProducer : public edm::one::EDAnalyzer<>
+{
+	public:
+		explicit DTCCablingMapProducer(const edm::ParameterSet&);
+		~DTCCablingMapProducer() override;
+		
+		static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+		
+	private:
+		void beginJob() override;
+		void analyze(const edm::Event&, const edm::EventSetup&) override;
+		void endJob() override;
+		
+		virtual void LoadModulesToDTCCablingMapFromCSV(std::vector<std::string> const&);
+		
+	private:
+		bool                                            generate_fake_valid_gbtlink_and_elinkid_;
+		int                                             verbosity_                              ;
+		unsigned                                        csvFormat_ncolumns_                     ;
+		unsigned                                        csvFormat_idetid_                       ;
+		unsigned                                        csvFormat_idtcid_                       ;
+		unsigned                                        csvFormat_igbtlinkid_                   ;
+		unsigned                                        csvFormat_ielinkid_                     ;
+		cond::Time_t                                    iovBeginTime_                           ;
+		std::unique_ptr<TrackerDetToDTCELinkCablingMap> pCablingMap_                            ;
+		std::string                                     record_                                 ;
+};
+
+
+void DTCCablingMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+	edm::ParameterSetDescription desc;
+	desc.setComment("Stores a TrackerDetToDTCELinkCablingMap object into the database from a CSV file.");
+	desc.add<bool>("generate_fake_valid_gbtlink_and_elinkid", false);
+	desc.add<int>("verbosity", 0);
+	desc.add<unsigned>("csvFormat_idetid"           ,  0);
+	desc.add<unsigned>("csvFormat_ncolumns"         , 15);
+	desc.add<unsigned>("csvFormat_idtcid"           , 10);
+	desc.add<unsigned>("csvFormat_igbtlinkid"       , 11);
+	desc.add<unsigned>("csvFormat_ielinkid"         , 12);
+	desc.add<long long unsigned int>("iovBeginTime" ,  1);
+	desc.add<std::string>("record","TrackerDTCCablingMapRcd");
+	desc.add<std::vector<std::string>>("modulesToDTCCablingCSVFileNames",std::vector<std::string>());
+	descriptions.addDefault(desc);
```

Comment:
@luigicalligaris I think here instead that
```C++
descriptions.addDefault(desc);
```
you should use:
```C++
 descriptions.add("DTCCablingMapProducer", desc);
```
at that point you can just put in the configuration file
```python
process.load("CondTools.SiPhase2Tracker.DTCCablingMapProducer")
```
without the need for creating a cfi file.

Reply:
done, pushed with the last commit